### PR TITLE
Format quota email currency values

### PR DESCRIPTION
**Pull request by JoaoJandre**
### Description

Currently, in the Quota emails, the formatting of monetary values does not include the currency symbol defined in the `quota.currency.symbol` configuration. Additionally, the values are not formatted.

This PR changes the Quota emails to use the quota.currency.symbol configuration to set the currency symbol. Additionally, a new configuration has also been created, `quota.currency.locale`, which should be set to format the value's punctuation in the quota emails using the wanted locale, en-US or pt-BR for example.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### How Has This Been Tested?
I tested different combinations to observe the changes in the received quota emails.

Test value = 12863.56
| `quota.currency.symbol` | `quota.currency.locale` | How it is received in the emails |
|-------------------------|-------------------------|---------------------------|
| R$ 					  | pt-BR				    | R$ 12.863,56		        |
| test 					  | `null`				    | test 12863.56		        |
| `null`				  | ja-JP				    | $ 12,863.56		        |

### CODE REVIEW

File: framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java [hunk 1]
```diff
@@ -208,8 +212,21 @@ public void sendQuotaAlert(DeferredQuotaEmail emailToBeSent) {
...
-            final Map<String, String> bodyOptionMap = generateOptionMap(account, userNames, accountDomain, balance, usage, emailType, true);
+            String currencyLocale = _configDao.getValue(QuotaConfig.QuotaCurrencyLocale.key());
+            String currencySymbol = ObjectUtils.defaultIfNull(_configDao.getValue(QuotaConfig.QuotaCurrencySymbol.key()), QuotaConfig.QuotaCurrencySymbol.defaultValue());
+
+            NumberFormat localeFormat = null;
+
+            if (currencyLocale != null) {
+                Locale locale = Locale.forLanguageTag(currencyLocale);
+                localeFormat = NumberFormat.getNumberInstance(locale);
+            }
```
Comment (DaanHoogland) [inline framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java:223]: can you extract this?

**Review by DaanHoogland — APPROVED** [review]
good functionality, small remarks

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by JoaoJandre** [comment]
@DaanHoogland could we rerun the tests for this PR?

**Comment by JoaoJandre** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: utils/src/main/java/com/cloud/utils/NumbersUtil.java [hunk 2]
```diff
@@ -147,6 +149,26 @@ public static long parseInterval(String str, long defaultValue) {
...
     }
 
+    /**
+     * Formats a BigDecimal with the number formatter.
+     * @return
+     * <li>null if value is null;</li>
+     * <li>value.toString() if the numberFormat is null;</li>
+     * <li>the value formatted if both parameters are not null;</li>
+     */
+    public static String formatBigDecimalAccordingToNumberFormat(BigDecimal value, NumberFormat numberFormat) {
```
Comment (GutoVeronezi) [inline utils/src/main/java/com/cloud/utils/NumbersUtil.java:159]: We could add some unit tests for this method.

File: framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java [hunk 3]
```diff
@@ -238,19 +248,29 @@ public void sendQuotaAlert(DeferredQuotaEmail emailToBeSent) {
         }
     }
 
+    private NumberFormat getLocaleFormatIfCurrencyLocaleNotNull() {
```
Comment (GutoVeronezi) [inline framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java:251]: We could add some unit tests for this method.

**Review by GutoVeronezi — APPROVED** [review]
CLGTM

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Pull request merged**